lightning/config/config.go

```diff
@@ -204,8 +205,58 @@ func (cfg *Config) LoadFromGlobal(global *GlobalConfig) error {
 }
 
 // LoadFromTOML overwrites the current configuration by the TOML data
+// If data contains toml items not in Config and GlobalConfig, return an error
+// If data contains toml items not in Config, thus won't take effect, warn user
 func (cfg *Config) LoadFromTOML(data []byte) error {
-	return errors.Trace(toml.Unmarshal(data, cfg))
+	// warn legal toml items but won't effect
+	var warnItems []string
+	dataStr := string(data)
+	metaData, err := toml.Decode(dataStr, cfg)
+
```

Comment:
throw error here?

Reply:
`err` is kept and returned at last lines of this function. I referenced [this piece](https://github.com/pingcap/tidb/blob/ddb6132e82f79a91518d75c91617f3d60d0de06b/config/config.go#L527-L538). But in fact now I prefer check `err` here.